include/irr/config/BuildConfigOptions.h.in

```diff
@@ -3,6 +3,9 @@
 
 #cmakedefine _IRR_DEBUG
 
+// working root directory
+#cmakedefine __IRR_ROOT_DIRECTORY__ "${CMAKE_SOURCE_DIR}"
```

Comment:
you want `_IRR_INSTALL_DIRECTORY_` and you should be getting the CMAKE installation directory, I will fix it myself afterwards actually

Reply:
ok this is actually quite a bit complex, but lets do it like this:

1.
When the Irrlicht targets are not installed (all examples and lib run out of the build-dir), this configured header (whose configured version will be placed in `{binary_dir}/include/irr/config/BuildConfigOptions.h` as it is right now, but unlike how it is now, NOT ADDED to the INSTALL target) 

NEEDS to evaluate to `#define _IRR_INSTALL_DIRECTORY_  "{binary_dir}"`

2. When the IRrlicht targets are installed (all examples run out of somewhere, and the library and includes are installed somewhere in a system-wide directory) this configured header will probably have to be placed somewhere out of the way, but added to the INSTALL target.

and the macro NEEDS to evaluate to `#define _IRR_INSTALL_DIRECTORY_ "${CMAKE_INSTALL_PREFIX}"`

Reply:
Are the files with shader code copied to the install dir as is now?

Reply:
you should check what the result (resulting files, directories, etc.) is when actually going through with INSTALLing to `Program Files`

Reply:
actually scap this idea, we don't need this define here